File: src/pi/pi.cpp
#include <iostream>
using namespace std;

double arc(double x);
const long int num_steps=1000000000;
double step;
int main()
{
  double pi, sum=0.0;
  step=1.0/(double) num_steps;

#pragma omp parallel for reduction(+:sum)
  for (int i=0;i<=num_steps;i++)
  {
    double x=(i+0.5)*step;
    sum+= arc(x);
  }
  pi = sum*step;

  cout.precision(10);
  cout << "pi is probably " 
    << fixed << pi << endl;

  return 0;
}
double arc(double x)
{
  double y = 4.0/(1+x*x);
  return y;
}

File: src/pi/pi.f90
program integratePi

  implicit none
  integer(kind=8) :: num_steps,i
  real(kind=8) :: sum,step,pi,x

  num_steps=1000000000
  sum=0.d0
  step=1.d0/num_steps

  call helloworld

  !$omp parallel do &
  !$omp private(x) &
  !$omp reduction(+:sum)
  do i=1,num_steps
    x=(i+0.5d0)*step
    sum=sum+arc(x)
  enddo
  !$omp end parallel do


  pi=sum*step

  write(6, &
   '("pi is probably ",f12.10)') &
    pi

  contains
    function arc(x)
      implicit none
      real(kind=8) :: arc,x
      arc = 4.d0/(1.d0+x*x)
    end function arc

end program integratePi

subroutine hello.world
  implicit none
  write(6,*) 'hello'
end subroutine hello.world
